src/main/java/teammates/ui/controller/ControllerServlet.java

```diff
@@ -133,9 +132,14 @@ public final void doPost(HttpServletRequest req, HttpServletResponse resp) throw
             EmailWrapper errorReport =
                     new EmailGenerator().generateSystemErrorEmail(requestMethod, requestUserAgent, requestPath,
                                                                   requestUrl, requestParams, userType, t);
-            new EmailSender().sendReport(errorReport);
-            log.severe(ActivityLogEntry.generateSystemErrorReportLogMessage(req, errorReport, userType));
-
+            /*
+             * Error report is not sent immediately as it can delay termination of request which may result
+             * in GAE shutting down the instance. The error report will be sent by email in the cron job
+             * auto/compileLogs later
+             */
+            if (errorReport != null) {
+                log.severe(ActivityLogEntry.generateSystemErrorReportLogMessage(req, errorReport, userType));
+            }
```

Comment:
As we are trying to save time here (we have less than 1 second to finish this part), perhaps just a simple `log.severe("Unexpected exception " + e.getMessage())` is enough, in which case we can avoid lines 126-134 too?